share/github-backup-utils/ghe-restore-repositories-gist

```diff
@@ -57,15 +59,65 @@ echo "$config" > "$config_file"
 
 opts="$GHE_EXTRA_SSH_OPTS -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PasswordAuthentication=no"
 
+# Remote sync-in-progress file location. When this file exists, Git GC
+# operations are disabled on the GitHub instance.
+sync_in_progress_file="$GHE_REMOTE_DATA_USER_DIR/repositories/.sync_in_progress"
+
 cleanup() {
   for pid in $(jobs -p); do
     kill -KILL $pid > /dev/null 2>&1 || true
   done
+  for hostname in $hostnames; do
+    ghe-ssh -F $config_file "$hostname:$port" -- "sudo rm -f '$sync_in_progress_file'"
+  done
   rm -rf $config_file ssh_routes_in ssh_routes_out ssh_finalize_in ssh_finalize_out
 }
 
 trap 'cleanup' INT TERM EXIT
 
+wait_pids=""
+for hostname in $hostnames; do
+  # Touch the sync-in-progress file, disabling GC operations, and wait for all
+  # active GC processes to finish on the remote side.
+  echo "
+      set -e
+      sudo -u git touch '$sync_in_progress_file'
+      sanity=0
+      while [ \$sanity -lt $GHE_GIT_COOLDOWN_PERIOD ]; do
```

Comment:
`for i in $(seq $GHE_GIT_COOLDOWN_PERIOD); do` may help to avoid using the `sanity` counter. Am I wrong?